core/src/price.ts

```diff
@@ -49,3 +49,39 @@ export const calculateTransactionProfit = function (auction: Auction): BigNumber
     const totalMarketPriceLimitedByDebt = collateralAmountLimitedByDebt.multipliedBy(auction.marketUnitPrice);
     return totalMarketPriceLimitedByDebt.minus(auction.debtDAI);
 };
+
+export const calculateTransactionProfitDate = function (auction: Auction): Date | undefined {
+    if (
+        auction.secondsBetweenPriceDrops === undefined ||
+        auction.secondsTillNextPriceDrop === undefined ||
+        auction.marketUnitPrice === undefined ||
+        auction.priceDropRatio === undefined
+    ) {
+        return undefined;
+    }
+
+    const isAlreadyProfitable = auction.approximateUnitPrice.isLessThan(auction.marketUnitPrice);
+    const date = new Date();
+
+    let steps = 0;
+    let currentValue = auction.approximateUnitPrice;
```

Comment:
Since `auction.approximateUnitPrice` is a bignumber and bignumber is an object, [`=` operator doesn't copy the value](https://www.digitalocean.com/community/tutorials/copying-objects-in-javascript), but only copy the reference to that object. Which means modifying `currentValue` will still modify `auction.approximateUnitPrice`.

Big number library actually have a special method for that: https://mikemcl.github.io/bignumber.js/#clone

Reply:
Yes, you're right! `new BigNumber` seems to be the correct way. Sorry for the wrong method suggestion